waiter/src/waiter/core.clj

```diff
@@ -611,16 +611,21 @@
 
 (def scheduler
   {:scheduler (pc/fnk [[:curator leader?-fn]
-                       [:settings scheduler-config]
-                       [:state service-id-prefix]
+                       [:settings scheduler-config scheduler-syncer-interval-secs]
+                       [:state scheduler-state-chan service-id-prefix]
                        service-id->password-fn*
-                       service-id->service-description-fn*]
+                       service-id->service-description-fn*
+                       start-scheduler-syncer-fn]
                 (let [is-waiter-app?-fn (fn is-waiter-app? [^String service-id]
                                           (str/starts-with? service-id service-id-prefix))
                       scheduler-context {:is-waiter-app?-fn is-waiter-app?-fn
                                          :leader?-fn leader?-fn
+                                         :scheduler-state-chan scheduler-state-chan
+                                         ;; TODO scheduler-syncer-interval-secs should be inside the scheduler's config
```

Comment:
I did not move the `scheduler-syncer-interval-secs` inside each individual scheduler config yet as I wanted to get a clearance on this PR first. Once this PR is in, I will make a follow-up PR with that change.